llvm/lib/Transforms/InstCombine/InstCombineAddSub.cpp

```diff
@@ -2742,6 +2742,17 @@ Instruction *InstCombinerImpl::visitSub(BinaryOperator &I) {
   return TryToNarrowDeduceFlags();
 }
 
+static FastMathFlags combineFastMathFlagsForFNeg(FastMathFlags FMF,
```

Comment:
Name is too specific, but I'm not sure what to call this. Also, it should be directly in FastMathFlags.

Maybe unionWithRewriteIntersect? Plus the no infinity special case can be applied after

Reply:
Can we try this approach in `FMF`?
```cpp
    FastMathFlags Result = this->Flags & Other.Flags;
    Result.setNoNaNs(this->noNaNs() || Other.noNaNs());
    Result.setNoSignedZeros(this->noSignedZeros() || Other.noSignedZeros());
    return Result;
```